gtsfm/densify/interface_patchmatchnet.py

```diff
@@ -0,0 +1,178 @@
+"""Interface class from gtsfmData to patchmatchnetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.densify.mvs_math import piecewise_gaussian
+
+NUM_PATCHMATCHNET_STAGES = 4
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for Patchmatch Net. It contains the interface from GtsfmData"""
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Initialize method for PatchmatchnetData
+
+        Args:
+            images: input images (H, W, C) to GTSFM
+            sfm_result: sfm results calculated by GTSFM
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
+        """
+        assert images is not None and len(images) > 1
+
+        # cache sfm result
+        self.sfm_result = sfm_result
+
+        # Patchmatch Net meta
+        self.num_views = num_views
+        self.num_stages = NUM_PATCHMATCHNET_STAGES
+
+        # Test data preparation
+        self.keys = sorted(self.sfm_result.get_valid_camera_indices())
+        self.num_images = len(self.keys)
+        self.keys_map = {}
+        for i in range(self.num_images):
+            self.keys_map[self.keys[i]] = i
+        self.images = images
+        self.image_w = images[self.keys[0]].width
+        self.image_h = images[self.keys[0]].height
+
+        self.pairs, self.depth_ranges = self.configure()
+
+    def configure(self) -> Tuple[np.ndarray, np.ndarray]:
+        """Configure pairs and depth_ranges for each view from sfm_result
```

Comment:
Why do we need this function?

Reply:
I think the initialization method might be too long and complicated without this function...

Reply:
I'm all for it, but I just don't understand it's purpose. Is it filtering away implausible pairs? Or how would you describe the algorithm in 1 sentence?

Reply:
if there are 12 valid images and the patchmatchnet's number of views is 5, the configure function does:

Reply:
cool, can you put that into the docstring? it's quite helpful